transformations/slangificator/transformation.py

```diff
@@ -0,0 +1,217 @@
+import itertools
+import random
+import spacy
+import os
+
+
+
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+
+"""
+Base Class for implementing the different input transformations a generation should be robust against.
+"""
+
+
+def slangify(self, text, probReplaceNoun=0.5, probReplaceAdjective=0.5, probReplaceAdverb=0.5, seed=0, max_outputs=1):
+    random.seed(seed)
+
+    #probReplaceNoun=0.5 # Probability to replace noun, which is in the dictionary
+    #probReplaceAdjective=0.5 # Probability to replace noun, which is in the dictionary
+    #probReplaceAdverb=0.5 # Probability to replace noun, which is in the dictionary
+    
+    nlp = spacy.load("en_core_web_sm") # get an instnace of the tikenizer
+    perturbed_texts = [] # output for all perturbed texts
+    
+    # Load dictionaries
+    Slang_Nouns =self.Slang_Nouns
+    Slang_Nouns = list(map(list, zip(*Slang_Nouns)))
+    Slang_Adverbs =self.Slang_Adverbs
+    Slang_Adverbs = list(map(list, zip(*Slang_Adverbs)))
+    Slang_Adjectives =self.Slang_Adjectives
+    Slang_Adjectives = list(map(list, zip(*Slang_Adjectives)))    
+    
+    # Tags for nouns
+    noun_tag=['NN', 'NNS', 'NNPS', 'NNP']
+    
+
+    
+    #Tokenize text
+    doc = nlp(text)    
+
+
+    
+    for _ in itertools.repeat(None, max_outputs):
+        
+
+        ReplPot=0 # counts potential replcacements, which could have been made if the probability of replcement would be set to one for all PoS
+        ReplMade=0 # counts actual replcacements
+        
+        modified_toks=[] # modified tokens
+        for token in doc: 
+            isCap=token.text[0].isupper() # check if token begins with the capital letter
+            
+            #Nouns
+            if token.tag_ in noun_tag:
+                #Check if word is in the corresponding dictionary
+                if token.lemma_ in Slang_Nouns[0]:   
+                    ReplPot+=1 # increment potential replacements
+                    
+                    repDecision=random.uniform(0, 1)<=probReplaceNoun # Randomly decide whether to replace or not
+                    if repDecision: #if replcement is made
+                        ReplMade+=1 
+                        
+                        #Choose a new word for replacement
+                        #ind=Slang_Nouns[0].index(token.lemma_)   
+                        indAll = [i for i, x in enumerate(Slang_Nouns[0]) if x == token.lemma_] # all possible replacements 
```

Comment:
Can you used more telling variable named than `ind`, `ind_r` and `indAll`? It's a bit difficult to follow

Reply:
Made new names to these variables in the revision